### PR TITLE
Ensure headless service port equal to targetPort

### LINKED ISSUE #17634: Service Port validation for headless services should ensure port == targetPort
Since headless services do not have any port-remap capabilities, we should enforce that the user did not ask for any remapping.  relatively easy fix.

### PR DESCRIPTION
**Pull request by feihujiang**
Fix #17634

### CODE REVIEW

**Comment by bgrant0607** [comment]
Thanks. Assigning to @thockin to choose a reviewer.

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ const isNegativeErrorMsg string = `must be non-negative`
 const fieldImmutableErrorMsg string = `field is immutable`
 const cIdentifierErrorMsg string = `must be a C identifier (matching regex ` + validation.CIdentifierFmt + `): e.g. "my_name" or "MyName"`
 const isNotIntegerErrorMsg string = `must be an integer`
+const headlessServicePortErrorMsg string = `headless service port must equal to targetPort`
```
Comment (thockin) [inline pkg/api/validation/validation.go]: "must be equal"

Comment (thockin) [inline pkg/api/validation/validation.go]: actually, the while string should be "must be equal to targetPort when clusterIP = None"

Comment (thockin) [inline pkg/api/validation/validation.go]: in fact, no need to make this a constant - it's only used once.

**Comment by thockin** [comment]
just string nits then LGTM.

**Comment by feihujiang** [comment]
@k8s-bot test this please

**Comment by feihujiang** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by feihujiang** [comment]
@k8s-bot unit test this please

**Comment by feihujiang** [comment]
@k8s-bot test this please

**Comment by liggitt** [comment]
This breaks old (arguably incorrect) service definitions. At least worth a release note

**Comment by liggitt** [comment]
Prior to this, targetPort was ignored on headless service definitions?

**Comment by liggitt** [comment]
Actually, I think we should force the target port to match the port for headless services. If we'd caught this prior to a release, we could reject in validation, but I'd consider this compatibility issue now.

**Comment by liggitt** [comment]
removing the release note, backing out the compatibility issue in https://github.com/kubernetes/kubernetes/pull/21680

**Pull request merged**